.github/workflows/issue-management.yml

```diff
@@ -0,0 +1,87 @@
+name: Issue Management
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # runs daily at midnight UTC
+  workflow_dispatch: # allows you to manually trigger the action
+
+jobs:
+  manage-issues:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Manage inactive issues
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const twoWeeksAgo = new Date(Date.now() - 14 * 24 * 60 * 60 * 1000);
+            const threeWeeksAgo = new Date(Date.now() - 21 * 24 * 60 * 60 * 1000);
+            const botLogin = 'github-actions[bot]';
+
+            const issues = await github.rest.issues.listForRepo({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              state: 'open'
+            });
+
+            for (const issue of issues.data) {
+              const comments = await github.rest.issues.listComments({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: issue.number
+              });
+
+              const lastNonBotActivity = new Date(Math.max(
+                ...comments.data
+                  .filter(comment => comment.user.login !== botLogin)
+                  .map(comment => new Date(comment.created_at)),
+                new Date(issue.created_at)
+              ));
+
+              const isAssigned = issue.assignees.length > 0;
+              const warningAlreadyGiven = comments.data.some(comment =>
+                comment.user.login === botLogin &&
+                comment.body.includes('This issue has been inactive')
+              );
+
+              if (lastNonBotActivity < threeWeeksAgo) {
+                // handle issues inactive for 3 weeks
+                if (isAssigned) {
+                  await github.rest.issues.removeAssignees({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    issue_number: issue.number,
+                    assignees: issue.assignees.map(a => a.login)
+                  });
+
+                  await github.rest.issues.createComment({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    issue_number: issue.number,
+                    body: 'This issue has been inactive for 3 weeks and has been unassigned. Feel free to pick it up again when you\'re ready to work on it.'
+                  });
+                } else if (!warningAlreadyGiven) {
+                  await github.rest.issues.createComment({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    issue_number: issue.number,
+                    body: 'This unassigned issue has been inactive for 3 weeks. Please consider assigning it to someone or updating its status.'
+                  });
+                }
```

Comment:
I appreciate you looking ahead, but I think we can remove this case. We sometimes create top-level "project" issues with a bunch of subtasks and we don't necessarily want to assign the top-level issue to anyone, so would be a tad annoying to get pinged about this.

Also, like you mentioned also, issues go very fast around here! 😂